src/main/java/com/cloudbees/plugins/credentials/CredentialsProviderTypeRestriction.java

```diff
@@ -128,13 +127,8 @@ public Includes(String provider, String type) {
         }
 
         private Object readResolve() {
-            Descriptor<?> descriptor = Jenkins.get().getDescriptor(type);
-            if (descriptor != null) {
-                return this;
-            }
-
-            this.type = convertDescriptorClassNameToId(type);
-
+            provider = convertDescriptorClassNameToId(provider);
```

Comment:
Also adding conversion of `provider` here, though given https://github.com/jenkinsci/credentials-plugin/blob/92471280cda18d1d980799080b3dcfa09fb70f8b/src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java#L1027-L1033 the mistake would usually pass unnoticed—the strings would normally be the same. That is also why users reported a regression in `CredentialsTypeFilter` and not `CredentialsProviderFilter` even though it was also incorrect.